app/models/env_measurement.rb

```diff
@@ -1,3 +1,6 @@
 class EnvMeasurement < ApplicationRecord
 	belongs_to :garden
+
+	validates_numericality_of :soil_temperature
+	validates_numericality_of :soil_moisture, greater_than_or_equal_to: 0
```

Comment:
Good thinking preventing the negative percentage.  I also like the numerical temperature validation.